### PR TITLE
Backend/payout-management-dashboard

### PR DESCRIPTION
**Pull request by prashant601**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/Storage/Beam/RiderDetails.hs [hunk 1]
```diff
@@ -15,26 +15,27 @@ import qualified Kernel.Types.Common
 import Tools.Beam.UtilsTH
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/Storage/Beam/RiderDetails.hs:15]: Please run runner with hlint.

File: Backend/dev/migrations/dynamic-offer-driver-app/0484-indexing-on-rider-details.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+CREATE INDEX idx_ride_id ON atlas_driver_offer_bpp.rider_details USING btree (first_ride_id);
```
Comment (hkmangla) [inline Backend/dev/migrations/dynamic-offer-driver-app/0484-indexing-on-rider-details.sql:1]: 😢

**Review by hkmangla — DISMISSED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/Storage/Beam/RiderDetails.hs (hunk 1 above)
Comment (prashant601) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/Storage/Beam/RiderDetails.hs]: done

**Review by khuzema786 — APPROVED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/Payout.hs [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+  mapM_ (callPayoutAndUpdateDailyStats merchant merchantOpCity) payoutOrders
+  pure Success
+
+callPayoutAndUpdateDailyStats :: Domain.Types.Merchant.Merchant -> DMOC.MerchantOperatingCity -> PO.PayoutOrder -> Environment.Flow ()
+callPayoutAndUpdateDailyStats merchant merchantOpCity payoutOrder = do
+  let driverId = Id payoutOrder.customerId
+  when (isNothing payoutOrder.vpa) $ throwError $ InvalidRequest "VPA does not Exist" -- vpa will always exist
+  orderStatusRep <- getPayoutOrderStatus (driverId, merchant.id, merchantOpCity.id) payoutOrder
+  when (orderStatusRep.status `elem` [TPayout.FULFILLMENTS_FAILURE, TPayout.FULFILLMENTS_CANCELLED, TPayout.FAILURE, TPayout.ERROR]) do
+    mbVehicle <- QVeh.findById (cast driverId)
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/Payout.hs:248]: Most of this code is already there and can be reused.

File: Backend/lib/payment/src-read-only/Lib/Payment/Storage/Queries/PayoutOrder.hs [hunk 4]
```diff
@@ -24,6 +24,9 @@ create = createWithKV
 createMany :: (Lib.Payment.Storage.Beam.BeamFlow.BeamFlow m r) => ([Lib.Payment.Domain.Types.PayoutOrder.PayoutOrder] -> m ())
 createMany = traverse_ create
 
+findAllWithStatus :: (Lib.Payment.Storage.Beam.BeamFlow.BeamFlow m r) => (Kernel.External.Payout.Juspay.Types.Payout.PayoutOrderStatus -> m [Lib.Payment.Domain.Types.PayoutOrder.PayoutOrder])
+findAllWithStatus status = do findAllWithKV [Se.Is Beam.status $ Se.Eq status]
```
Comment (hkmangla) [inline Backend/lib/payment/src-read-only/Lib/Payment/Storage/Queries/PayoutOrder.hs:28]: Its better to have some limit and offset in this else can get whole table data.

**Pull request merged**